common-pg/src/main/scala/com/socrata/pg/soql/SqlFunctions.scala

```diff
@@ -620,8 +619,14 @@ object SqlFunctions extends SqlFunctionsLocation with SqlFunctionsGeometry with
       case x => x
     }
 
+    val litCtx = soqlType match {
+      case SoQLFloatingTimestamp | SoQLFixedTimestamp =>
+        ctx + (SqlizerContext.TimestampLiteral -> true)
+      case _ =>
+        ctx
+    }
     val fnWithConvertedParams = fn.copy(parameters = convertedParams, window = fn.window)
-    formatCall(template, None, None)(fnWithConvertedParams, rep, typeRep, setParams, ctx, escape)
+    formatCall(template, None, None)(fnWithConvertedParams, rep, typeRep, setParams, litCtx, escape)
```

Comment:
It feels like it'd be cleaner just to do it here?

```scala
match (soqlType, convertedParams) {
  case (SoQLFloatingTimestamp, StringLiteral(SoQLFloatingTimestamp.StringRep(ts), _)) =>
    // at this point ts is a LocalDateTime and we can generate a ParametricSql that is just that
  case (SoQLFixedTimestamp, StringLiteral(SoQLFixedTimstamp.StringRep(ts), _)) =>
    // at this point ts is a DateTime and we can generate a ParametricSql that is just that
  case _ =>
      ... old codegen here ...
}
```

Reply:
Can't share much of the crazy code (like logic to not use parameter if it is in group by) if it is done here.

Reply:
like this:
select count(id),
round((extract(epoch from created_at) - extract(epoch from '2012-01-01'::timestamp with time zone))::numeric/86400)
from lenses
group by round((extract(epoch from created_at) - extract(epoch from '2012-01-01'::timestamp with time zone))::numeric/86400)